source/geod24/LocalRest.d

```diff
@@ -303,10 +303,12 @@ private class Connection
             Response res;
             scheduler.start({
                 // Loop until we get the Response we are looking for
-                while (this.resp_chn.read(res, timeout))
-                    if (res.id == resp_id)
-                        return;
-                res = Response(Status.Timeout, resp_id, SerializedData("Request timed-out"));
+                if (!this.resp_chn.read(res, timeout))
+                    res = Response(Status.Timeout, resp_id, SerializedData("Request timed-out"));
+                // Otherwise the `res` variable would have been filled by `read`,
+                // and the response ID should match since the main thread never
+                // send concurrent requests
+                assert(res.id == resp_id);
```

Comment:
Could we hit a case similar to what we encountered in `vibe.d`?
A response for a timed out request could be waiting in the channel. Probably will never happen in practice.

Reply:
Hum that's actually a good point, I'll look at the code once more, and will take this commit out in the meantime.